chainer/links/activation/maxout.py

```diff
@@ -59,19 +60,46 @@ def __init__(self, in_size, out_size, pool_size,
         super(Maxout, self).__init__()
 
         linear_out_size = out_size * pool_size
-        if initialW is not None:
-            initialW = initialW.reshape(linear_out_size, in_size)
 
-        if initial_bias is not None:
-            if numpy.isscalar(initial_bias):
-                dtype = chainer.get_dtype()
-                initial_bias = numpy.full(
-                    (linear_out_size,), initial_bias, dtype=dtype)
-            elif isinstance(initial_bias, (numpy.ndarray, cuda.ndarray)):
-                initial_bias = initial_bias.reshape(linear_out_size)
-            else:
-                raise ValueError(
-                    'initial bias must be float, ndarray, or None')
+        if numpy.isscalar(initialW):
+            pass
+        elif isinstance(initialW, (numpy.ndarray, cuda.ndarray)):
+            if initialW.ndim != 3:
+                raise ValueError('initialW.ndim should be 3')
+            initialW = initialW.reshape(linear_out_size, in_size)
+        elif isinstance(initialW, initializer.Initializer):
+            pass
+        elif callable(initialW):
+            def make_initialW_wrapper(initialW):
```

Comment:
I don't think we need such a rich func wrapper here. How about making it simpler as follows:
```py
        initialW_orig = initialW
        def initialW(array):
            array.shape = (out_size, pool_size, in_size)
            initialW_orig(array)
            array.shape = (linear_out_size, in_size)
```

Reply:
oh... fixed!